src/custom-apps/handlers.ts

```diff
@@ -24,14 +44,82 @@ export const createSyncEventsSelectHandler = (
 export const createAsyncEventsSelectHandler = (
   change: (event: ChangeEvent, cb?: () => void) => void,
   asyncEvents: WebhookEventTypeAsyncEnum[],
+  query: string,
+  setQuery: React.Dispatch<React.SetStateAction<string>>,
 ) => (event: ChangeEvent) => {
   const events = toggle(event.target.value, asyncEvents, (a, b) => a === b);
   const filteredEvents = filterSelectedAsyncEvents(events);
 
+  // Clear syncEvents
+  change({
+    target: {
+      name: "syncEvents",
+      value: [],
+    },
+  });
+
   change({
     target: {
       name: "asyncEvents",
       value: filteredEvents,
     },
   });
+
+  handleQuery(filteredEvents, query, setQuery);
+};
+
+const handleQuery = (
+  events: WebhookEventTypeAsyncEnum[],
+  query: string,
+  setQuery: React.Dispatch<React.SetStateAction<string>>,
+) => {
+  if (events.length > 0 && query.length === 0) {
+    const event = events[0]
+      .toLowerCase()
+      .split("_")
+      .map(chunk => capitalize(chunk))
+      .join("");
+    setQuery(
+      print(parse(`subscription { event { ... on ${event} { __typename } } }`)),
+    );
+  }
+
+  if (query.length > 0) {
+    const ast = parse(query);
+
+    visit(ast, {
+      SelectionSet(node, _key, parent) {
+        if ((parent as ObjectFieldNode).name?.value === "event") {
+          const queryEvents = node.selections.map(
+            selection =>
+              (selection as InlineFragmentNode).typeCondition.name.value,
+          );
+          const newEvents = events
+            .map(event =>
+              event
+                .toLowerCase()
+                .split("_")
+                .map(chunk => capitalize(chunk))
+                .join(""),
+            )
+            .filter(event => !queryEvents.includes(event));
+
+          if (newEvents.length > 0) {
+            // TODO modify AST
```

Comment:
👀

Reply:
@Droniu this is a workaround and we will improve it in the next iteration